### PR TITLE
Add tests for activity settings

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
The PR adds webUI acceptance tests activity settings.

## Related Issue
 #677 

## How Has This Been Tested?
:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

~~P.S: Some of the steps used here are yet to be merged in [PR: owncloud/core/pull/34294]~~

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 1]
```diff
@@ -108,6 +117,17 @@ public function theUserFiltersActivityListBy($activityType) {
 		$this->activityPage->filterActivityListBy($this->getSession(), $activityType);
 	}
 
+	/**
+	 * @When /^the user (disables|enables) activity log (stream|mail) for "([^"]*)" using the webUI$/
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]: params and returns..

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 2]
```diff
@@ -263,6 +283,18 @@ public function theActivityShouldNotHaveAnyMessageWithKeyword($tag) {
...
 
+	/**
+	 * @Then the activity list should be empty
+	 *
+	 * @return void
+	 */
+	public function theActivityListShouldBeEmpty() {
+		$activities = $this->activityPage->getAllActivityMessageLists();
+		PHPUnit_Framework_Assert::assertEmpty(
+			$activities, "Activity list was expected to be empyt but was not"
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]:
```suggestion
			$activities, "Activity list was expected to be empty but was not"
```

File: tests/acceptance/features/lib/ActivityForm.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Class ActivityForm
+ *
+ * @package Page
+ */
+class ActivityForm extends PersonalGeneralSettingsPage {
+	private $activityCheckboxXpath = "//td[contains(@data-select-group,'%s')]/preceding-sibling::td[%s]";
+	private $activityCheckboxId = "//input[@id='%s']";
+
+	public function changeActivityLogSetting(
```
Comment (dpakach) [inline tests/acceptance/features/lib/ActivityForm.php]: doc for this function

File: tests/acceptance/features/lib/ActivityForm.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author Paurakh Sharma Humagain <paurakh@jankaritech.com>
+ * @copyright Copyright (c) 2018, JankariTech
```
Comment (skshetry) [inline tests/acceptance/features/lib/ActivityForm.php]:
```suggestion
 * @copyright Copyright (c) 2019, JankariTech
```

File: tests/acceptance/features/lib/ActivityForm.php (hunk 3 above)
Comment (skshetry) [reply tests/acceptance/features/lib/ActivityForm.php]: This seems to be doing too much. Can we break this into smaller units?

File: tests/acceptance/features/lib/ActivityForm.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	 *
+	 * @param string $disablesOrEnables
+	 * @param string $streamOrMail
+	 * @param string $activityType
+	 * @param Session $session
+	 *
+	 * @return void
+	 * @throws \Exception
+	 */
+	public function changeActivityLogSetting(
```
Comment (skshetry) [inline tests/acceptance/features/lib/ActivityForm.php]: This seems to be doing too much. Can we break this into smaller units?

File: tests/acceptance/features/lib/ActivityForm.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+namespace Page;
+
+use Behat\Mink\Session;
+
+/**
+ * Class ActivityForm
+ *
+ * @package Page
+ */
+class ActivityForm extends PersonalGeneralSettingsPage {
```
Comment (skshetry) [inline tests/acceptance/features/lib/ActivityForm.php]: This is not an `ActivityForm`, is it?

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 7]
```diff
@@ -118,6 +128,23 @@ public function theUserFiltersActivityListBy($activityType) {
...
+	/**
+	 * @When /^the user (disables|enables) activity log (stream|mail) for "([^"]*)" using the webUI$/
+	 *
+	 * @param string $disablesOrEnables
+	 * @param string $streamOrMail
+	 * @param string $activityType
+	 *
+	 * @return void
+	 */
+	public function theUserDisablesActivityLogStreamOrMailForUsingTheWebui(
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]:
```suggestion
	public function theUseTriggersActivityLogSettingUsingTheWebui(
```

File: tests/acceptance/features/lib/ActivitySettingForm.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+			" cannot find label for $streamOrMail checkbox" .
+			"for $activityType activity"
+		);
+		$this->assertElementNotNull(
+			$checkCheckbox,
+			__METHOD__ .
+			" id $checkCheckboxFullId " .
+			"could not find checkbox"
+		);
+		if ($disablesOrEnables == 'enables') {
```
Comment (individual-it) [inline tests/acceptance/features/lib/ActivitySettingForm.php]:
```suggestion
		if ($disablesOrEnables === 'enables') {
```

File: tests/acceptance/features/lib/ActivitySettingForm.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+			if (!$checkCheckbox->isChecked()) {
+				$activityCheckbox->click();
+			}
+		} elseif ($disablesOrEnables == 'disables') {
```
Comment (individual-it) [inline tests/acceptance/features/lib/ActivitySettingForm.php]:
```suggestion
		} elseif ($disablesOrEnables === 'disables') {
```

**Comment by phil-davis** [comment]
@ownclouders rebase

**Pull request merged**